### DISCUSSION TITLE
How is prime p and the order of the finite field determined for ```thresha.random_split```

### DISCUSSION THREAD

**Discussion by RetoKrummenacher** [lschoe/mpyc#60] [Q&A] [answered]
Hi Berry,

Sorry for disturbing you again, but I was wondering how MPyC sets ```field.modulus``` and ```field.order``` used to create the Shamir Polynomials in ```thresha.random_split()```

I saw that they are set to be the same in ```finfields.pGF``` and that the prime $p$ is determined with ```find_prime_root(l)``` with bit length l=bit_length+sec_param, defaulting to $62=32+30$ as defined in the ```get_arg_parser()```.

But not sure if that is really the case?

Thanks for a clarification.

**Comment by lschoe** [comment]
So, `field.modulus` and `field.order` are in general not the same. This is only true for _prime_ order fields, which you are probably considering. For extension fields, we have that `field.modulus` is actually a polynomial (irreducible), and `field.order` is then a prime power (and there is also `field.characteristic`, which is always a prime number).

Functions like `mpc.SecFld()` and `mpc.SecInt()` create finite fields of the appropriate type, and the code for this is in the module `mpyc.sectypes`.

**Comment by RetoKrummenacher** [comment]
Thank you for the fast reply.

If I understand it correctly, if the parties are computing the sum of secure integers via `mpc.input(secint(integer))`, MPyC is using `mpc.SecInt()` as fields?

Those are *prime* order fields created with `sectypes.SecInt` calling `sectypes._pfield`, which finds a large prime number of given bit length via `sectypes.find_prime_root()` and creating the Galois field with `finfields.pGF()` where the order and the modulus are the same.
Those two are then used in `thresha.random_split()`?

**Reply by lschoe** [reply, marked as answer]
Right! So, if you set `secint = mpc.SecInt()`, then you can extract the corresponding field using `field = secint.field`. This will be a prime field, of which you can inspect `field.modulus` and `field.order`, which are both equal to the prime number used for secret sharing.

**Reply by RetoKrummenacher** [reply]
Thank you very much
